pytest_flask/fixtures.py

```diff
@@ -148,10 +151,10 @@ def test_server_is_up_and_running(live_server):
     host = pytestconfig.getvalue('live_server_host')
 
     # Explicitly set application ``SERVER_NAME`` for test suite
-    # and restore original value on test teardown.
     server_name = app.config['SERVER_NAME'] or 'localhost'
```

Comment:
For this simple case we can implement the "monkeypatch" functionality ourselves:

```python
    original_server_name = app.config.get('SERVER_NAME', None)
    final_server_name = _rewrite_server_name(original_server_name or 'localhost', str(port))
    app.config['SERVER_NAME'] = final_server_name
```

Then later, instead of `return server`, we write:

```python    
    yield server
    # restore original value on teardown
    if original_server_name is not None:
        app.config['SERVER_NAME'] = original_server_name
```

Reply:
Ok, this has been updated and now the test passes again. Thanks